cxx-sensors/src/test/java/org/sonar/cxx/sensors/tests/dotnet/WildcardPatternFileProviderTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2017 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.sensors.tests.dotnet;
+//origin https://github.com/SonarSource/sonar-dotnet-tests-library/
+
+import com.google.common.base.Joiner;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class WildcardPatternFileProviderTest {
+
+  @Rule
+  public TemporaryFolder tmp = new TemporaryFolder();
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Before
+  public void init() throws Exception {
+    tmp.newFile("foo.txt");
+    tmp.newFile("bar.txt");
+
+    tmp.newFolder("a");
+    tmp.newFile(path("a", "foo.txt"));
+    tmp.newFolder("a", "a21");
+
+    tmp.newFolder("b");
+
+    tmp.newFolder("c");
+    tmp.newFolder("c", "c21");
+    tmp.newFile(path("c", "c21", "foo.txt"));
+    tmp.newFolder("c", "c22");
+    tmp.newFolder("c", "c22", "c31");
+    tmp.newFile(path("c", "c22", "c31", "foo.txt"));
+    tmp.newFile(path("c", "c22", "c31", "bar.txt"));
+  }
+
+  @Test
+  public void absolute_paths() {
+    assertThat(listFiles(new File(tmp.getRoot(), "foo.txt").getAbsolutePath()))
+      .containsOnly(new File(tmp.getRoot(), "foo.txt"));
+
+    assertThat(listFiles(new File(tmp.getRoot(), path("c", "c22", "c31", "foo.txt")).getAbsolutePath()))
+      .containsOnly(new File(tmp.getRoot(), path("c", "c22", "c31", "foo.txt")));
+
+    assertThat(listFiles(new File(tmp.getRoot(), "nonexisting.txt").getAbsolutePath())).isEmpty();
+  }
+
+  @Test
+  public void absolute_paths_with_current_and_parent_folder_access() {
+    assertThat(listFiles(new File(tmp.getRoot(), path("a", "..", ".", "foo.txt")).getAbsolutePath()))
```

Comment:
@Bertk testing three times the same thing?

Reply:
This finding is correct. I removed the redundant statements.